### PR TITLE
shim-sev: migrated MxCsr type from crate xsave to x86_64

### LINKED ISSUE #1291: [Feature]: Use the MxCsr type from x86_64 crate
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Now that x86_64 for crate has MxCsr type it needs to be integrated to xsave and enarx subprojects.

### Acceptance Criteria

Use x86_64 MxCsr and make sure that this won't break any tests (including testing with SEV and SGX backends).

### Suggestions for a technical implementation

_No response_

### PR DESCRIPTION
**Pull request by mayankkumar2**
Migrated MxCsr type from xsave crate to x86_64 crate. 

Closes #1291

### CODE REVIEW

File: internal/shim-sev/src/sse.rs
Comment (haraldh) [inline internal/shim-sev/src/sse.rs:10]: You should run `cargo fmt` before submitting a PR. Or better automate your editor to do it on save for you.

```suggestion
use x86_64::registers::{
    mxcsr::MxCsr,
    xcontrol::{XCr0, XCr0Flags},
}
```

**Review by haraldh — CHANGES_REQUESTED** [review]
The correct patch for `internal/shim-sev/src/sse.rs` would be:
```diff
diff --git a/internal/shim-sev/src/sse.rs b/internal/shim-sev/src/sse.rs
index 3afdceaa..216ca092 100644
--- a/internal/shim-sev/src/sse.rs
+++ b/internal/shim-sev/src/sse.rs
@@ -5,10 +5,11 @@
 use crate::interrupts::XSAVE_AREA_SIZE;
 use crate::snp::cpuid_count;
 
-use core::arch::asm;
-
-use x86_64::registers::xcontrol::{XCr0, XCr0Flags};
-use xsave::MxCsr;
+use x86_64::registers::{
+    mxcsr,
+    mxcsr::MxCsr,
+    xcontrol::{XCr0, XCr0Flags},
+};
 
 /// Setup and check SSE relevant stuff
 pub fn init_sse() {
@@ -36,14 +37,12 @@ pub fn init_sse() {
     // Make sure that interrupts have enough room for xsave
     assert!(xsave_size <= XSAVE_AREA_SIZE);
 
-    unsafe {
-        let mxcsr: u32 = (MxCsr::INVALID_OPERATION_MASK
+    mxcsr::write(
+        MxCsr::INVALID_OPERATION_MASK
             | MxCsr::DENORMAL_MASK
             | MxCsr::DIVIDE_BY_ZERO_MASK
             | MxCsr::OVERFLOW_MASK
             | MxCsr::UNDERFLOW_MASK
-            | MxCsr::PRECISION_MASK)
-            .bits();
-        asm!("ldmxcsr [{}]", in(reg) &mxcsr, options(nostack));
-    }
+            | MxCsr::PRECISION_MASK,
+    );
 }
```

File: internal/shim-sev/src/sse.rs
Comment (mayankkumar2) [reply internal/shim-sev/src/sse.rs:10]: Sure, I will take care of this.

**Comment by jarkkojs** [comment]
A nit: `shim-sev:` would be more clear, i.e. a subsystem is not a feature even tho you're implementing a feature to a subsystem :-)

**Comment by mayankkumar2** [comment]
@jarkkojs yeah that makes sense, I will make the change.

**Comment by haraldh** [comment]
well the notation `feat(subsystem):` is quite common nowadays. see https://www.conventionalcommits.org/en/v1.0.0/

**Comment by mayankkumar2** [comment]
@haraldh so should I change the commit message? or is it fine?

**Comment by haraldh** [comment]
I am fine with both, since we are not enforcing one format yet.

**Review by haraldh — APPROVED** [review]

**Review by npmccallum — CHANGES_REQUESTED** [review]
I'm filing changes requested to block the merge. This PR adds a new git dependency just before a release. We should wait to merge this until after the release.

**Comment by npmccallum** [comment]
@mayankkumar2 Please rebase this PR. I have merged support for the latest release of `x86_64` in #1335. You should no longer need the git dependency.

**Comment by mayankkumar2** [comment]
@npmccallum thanks!

**Comment by mayankkumar2** [comment]
@npmccallum @jarkkojs @haraldh can you please review the PR.

**Review by haraldh — APPROVED** [review]

**Review by npmccallum — APPROVED** [review]

**Pull request merged**
